### ISSUE TITLE
Restarting training with slightly different params?  (Validation)

### ISSUE THREAD

**Issue by kellymarchisio** [marian-nmt/marian#263]
If I stop then restart a model during training with different validation parameters, am I going to cause a problem?   I've added translation to --valid-metrics, and added --valid-script-path --valid-translation-output and --quiet-translation. All other hyperparams should be the same. I figured I should ask before I accidentally overwrite my model, or something.  Thanks!

**Comment by frankseide** [comment]
Please protect against overwriting by copying the folder first :)

Then you can safely try it. I expect this to work, with one risk being the added metric. It is possible that it may want to access the previous value of that metric, and fail. If that happens, then please try locating the YML file in the model-output folder that contains the current state of the metrics, and manually add in an entry for the new metric.

In general,

If you interrupt and restart a model training, it will just restart where it stopped. Marian has no special intelligence to detect changes in hyperparameters. I.e, if you change hyperparameters, it will from that point on use those new hyperparameters.

Some hyperparams that change over time may do so based on the update count (stateless) and some based on updating a state. The latter will start from the last saved state.

Some hyperparams are related to optimizer state. E.g. if you change the Adam decay parameters, you may load an Adam state that is scaled wrongly.

I am not sure whether some hyperparameters are stored in the model file itself.

**Comment by kellymarchisio** [comment]
Thanks for the explanation - I'll give that a shot.
